sympy/tensor/array/array_comprehension.py

```diff
@@ -240,8 +240,8 @@ def _check_limits_validity(cls, function, limits):
             if any((not isinstance(i, Expr)) or i.atoms(Symbol, Integer) != i.atoms()
                                                                 for i in [inf, sup]):
                 raise TypeError('Bounds should be an Expression(combination of Integer and Symbol)')
-            if (inf > sup) == True:
-                raise ValueError('Lower bound should be inferior to upper bound')
+            if (inf > sup):
```

Comment:
This might actually change the semantics of the comparison: I think `inf` and `sup` might be `Symbol` instances, hence a `Gt` object might be formed so an actual check of equality of `True` might be what was intended here.

Reply:
Which explains this failing test instance:
```
__ sympy.tensor.array.array_comprehension.ArrayComprehension.is_shape_numeric __
File "/home/runner/work/sympy/sympy/sympy/tensor/array/array_comprehension.py", line 178, in sympy.tensor.array.array_comprehension.ArrayComprehension.is_shape_numeric
Failed example:
    b = ArrayComprehension(10*i + j, (i, 1, 4), (j, 1, k+3))
Exception raised:
    Traceback (most recent call last):
      File "/opt/hostedtoolcache/Python/3.10.8/x64/lib/python3.10/doctest.py", line 1350, in __run
        exec(compile(example.source, filename, "single",
      File "<doctest sympy.tensor.array.array_comprehension.ArrayComprehension.is_shape_numeric[5]>", line 1, in <module>
        b = ArrayComprehension(10*i + j, (i, 1, 4), (j, 1, k+3))
      File "/home/runner/work/sympy/sympy/sympy/tensor/array/array_comprehension.py", line 41, in __new__
        arglist.extend(cls._check_limits_validity(function, symbols))
      File "/home/runner/work/sympy/sympy/sympy/tensor/array/array_comprehension.py", line 243, in _check_limits_validity
        if (inf > sup):
      File "/home/runner/work/sympy/sympy/sympy/core/relational.py", line 511, in __bool__
        raise TypeError("cannot determine truth value of Relational")
    TypeError: cannot determine truth value of Relational
```